modules/dmrpp_module/ngap_container/NgapApi.cc

```diff
@@ -338,24 +338,93 @@ string get_data_access_url(rapidjson::Value &obj) {
     return {""};
 }
 
+
+string get_data_s3_url(rapidjson::Value &obj) {
```

Comment:
there is a little code duplication here, but not enough that it felt worth the time to refactor and make shared---especially since we'll likely be ripping out one of the three of these duplicate-ish functions in the near future anyway.